badc/source/main.rs

```diff
@@ -1,21 +1,44 @@
-use std::path::PathBuf;
+use std::{path::PathBuf, str::FromStr};
+use clap::Parser;
 
-use bad::CompilationConfiguration;
-use bad::ProgramSource;
+/// The badc compiler for the B language (Kernighan, 1969). Learning experiment for DrawsMiguel and ThePhD on Rust and some compilation techniques. Released un the CC0 1.0 Universal (e.g. Public Domain dedication).
+#[derive(Parser, Debug)]
+#[command(author, version, about = "A B language (Kernighan, 1969) compiler. Not at all useful.")]
+struct CommandLineCompilationOptions {
+	/// All of the paths to the input to compile, each one considered an independent translation unit.
+	#[arg(short, long)]
+	inputs: Vec<PathBuf>,
 
-pub struct CompilationOptions {
-	pub sources: Vec<ProgramSource>,
+	/// Print out the token sequence.
+	#[arg(short, long, default_value_t = true)]
+	print_tokens: bool,
+
+	/// Print out the AST.
+	#[arg(short, long, default_value_t = true)]
+	print_ast: bool,
+
+	/// The path to the output.
+	#[arg(short, long)]
+	output: Option<PathBuf>,
 }
 
 fn main() {
-	// TODO: parse from Command Line and/or JSON
-	let options = CompilationOptions {
-		sources: vec![ProgramSource::Path(PathBuf::from("./main.b"))],
-	};
-	// TODO: properly transfer compilation options to library configuration
-	let config = CompilationConfiguration {
-		sources: options.sources,
-	};
-	let tree = bad::compile(&config);
-	println!("{:?} 🎉!", tree);
+	let mut args = CommandLineCompilationOptions::parse();
+	if args.inputs.is_empty() {
+		args.inputs.push(PathBuf::from_str("./main.b").unwrap());
+	}
+	for input in args.inputs {
+		let output = match args.output {
+			Some(ref target_path) => target_path.clone(),
```

Comment:
`ref` should be redundant here?

Reply:
The error says:

> use of moved value
> move occurs because value has type `PathBuf`, which does not implement the `Copy` traitrustc[E0382](https://doc.rust-lang.org/error-index.html#E0382)
> main.rs(32, 9): borrow this field in the pattern to avoid moving `args.output.0`: `ref `

So I think I have to add the `ref`...

Reply:
Fixed it!!!